server/service/src/invoice/inbound_return/generate_lines.rs

```diff
@@ -0,0 +1,269 @@
+use crate::{service_provider::ServiceContext, ListError, ListResult};
+use repository::{
+    EqualFilter, InvoiceLineFilter, InvoiceLineRepository, ItemRow, RepositoryError, StockLineRow,
+};
+use util::uuid::uuid;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct InboundReturnLine {
+    pub id: String,
+    pub reason_id: Option<String>,
+    pub note: Option<String>,
+    pub number_of_packs: f64,
+    pub stock_line_row: StockLineRow,
+    pub item_row: ItemRow,
+    pub packs_issued: Option<f64>,
+}
+
+pub struct ExistingLinesInput {
+    pub item_id: String,
+    pub return_id: String,
+}
+
+pub struct GenerateInboundReturnLinesInput {
+    pub outbound_shipment_line_ids: Vec<String>,
+    pub existing_lines_input: Option<ExistingLinesInput>,
```

Comment:
Experimenting with this input... wanting to constrain:
- If you want to generate new lines, pass in outbound shipment lines to create them from
- If you want show the existing lines for a particular item, you're going to need to include both the item and return ids

Not sure how much value it ends up adding, because might be harder to handle on the frontend? Haven't got the frontend implementation of editing an existing inbound return in place yet so didn't get to test...

Also not sure about the naming, but interested in thoughts on this idea!

Reply:
> If you want show the existing lines for a particular item, you're going to need to include both the item and return ids

But, there will be an option to add outbound_shipment_line_id to existing return..

Reply:
I think maybe the `id_type` and `ids`, might be the way to go ?

Reply:
Yeah you might be right, I don't think there is a use case where we would supply both the outbound_shipment_line_id and the item_id? I'm kind of keen to get this in to unblock other work - perhaps we can revisit when we do the add to existing return piece?

Also just flagging here - if we want to determine whether an outbound_shipment_line_id is already in a return, we'll need to add a `linked_invoice_line_id` to InvoiceLineRow. With inbound_shipment/outbound_return we could check the stock_line_id, but with outbound_shipment/inbound_return we can't do this as new stock lines will be generated.

Reply:
> Also just flagging here - if we want to determine whether an outbound_shipment_line_id is already in a return, we'll need to add a linked_invoice_line_id to InvoiceLineRow

Yep agree

> unblock other work

Sure, would be great to see how it looks with the 'id_type' and 'id', which would translate to `enum ItemOrOutboundLineId { ItemId(String), OutboundLineId(String) }`